ChangeSet
  1.1542 04/03/31 14:01:03 dgibson@wocket.ozlabs.ibm.com +1 -0
  Fix stupid bug in iseries_veth driver.  Broken logic when splitting up the
  receive buffer for dma.

  drivers/net/iseries_veth.c
    1.17 04/03/31 14:00:30 dgibson@wocket.ozlabs.ibm.com +1 -1
    Fix stupid bug in iseries_veth driver.  Broken logic when splitting up the
    receive buffer for dma.

diff -Nru a/drivers/net/iseries_veth.c b/drivers/net/iseries_veth.c
--- a/drivers/net/iseries_veth.c	Wed Mar 31 06:30:04 2004
+++ b/drivers/net/iseries_veth.c	Wed Mar 31 06:30:04 2004
@@ -1132,7 +1132,7 @@
 	done = list[0].size;
 	while (done < length) {
 		list[i].addr = veth_dma_addr(p + done);
-		list[i].size = min(done, PAGE_SIZE);
+		list[i].size = min(length-done, PAGE_SIZE);
 		done += list[i].size;
 		i++;
 	}
.........................................................................
# vim: syntax=diff

